code/datums/elements/footstep.dm

```diff
@@ -125,44 +135,68 @@
 	if (forced || SHOULD_DISABLE_FOOTSTEPS(source) || !momentum_change)
 		return
 
+	var/list/prepared_steps = prepare_step(source)
+	if(isnull(prepared_steps))
+		return
+
+	var/footstep_type = null
+	var/list/footstep_sounds
+	var/stepcount = steps_for_living[source]
+	var/obj/item/bodypart/leg/left_leg = source.get_bodypart(BODY_ZONE_L_LEG)
+	var/obj/item/bodypart/leg/right_leg = source.get_bodypart(BODY_ZONE_R_LEG)
```

Comment:
These should be null and only be assigned values once the if statement for body parts covered doesn't pass. This obviously mean adding a layer of indentation but it's a mild opt.